src/storage.rs

```diff
@@ -70,30 +68,21 @@ impl Storage for GoStorage {
         end: Option<&[u8]>,
         order: Order,
     ) -> BackendResult<u32> {
-        // returns nul pointer in Buffer in none, otherwise proper buffer
-        let start_buf = start
-            .map(|s| Buffer::from_vec(s.to_vec()))
-            .unwrap_or_default();
-        let end_buf = end
-            .map(|e| Buffer::from_vec(e.to_vec()))
-            .unwrap_or_default();
```

Comment:
Before both `None` and vectors with capacity 0 turned into `nil`, because in both cases the pointer is null.